protocols/kad/src/protocol.rs

```diff
@@ -414,8 +502,24 @@ fn proto_to_resp_msg(mut message: proto::Message) -> Result<KadResponseMsg, io::
     match message.get_field_type() {
         proto::Message_MessageType::PING => Ok(KadResponseMsg::Pong),
 
-        proto::Message_MessageType::GET_VALUE =>
-            Err(invalid_data("Unsupported: GET_VALUE message")),
+        proto::Message_MessageType::GET_VALUE => {
+            let result = match message.has_record() {
+                true => {
+                    let mut record = message.take_record();
+                    let key = Multihash::from_bytes(record.take_key()).map_err(invalid_data)?;
+                    Some((key, record.take_value().into()))
+                }
+                false => None,
+            };
+
+            let closer_peers = message
+                .mut_closerPeers()
+                .iter_mut()
+                .filter_map(|peer| KadPeer::try_from(peer).ok())
```

Comment:
It may be preferable to fail here if serialisation of any `KadPeer` fails, instead of silently dropping entries that cannot be parsed, which may lead to confusion.

Reply:
but this is done in all other message types.